cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/DataSetServiceModules.java

```diff
@@ -69,6 +71,10 @@ protected void configure() {
         install(new FactoryModuleBuilder()
                   .implement(DatasetDefinitionRegistry.class, DefaultDatasetDefinitionRegistry.class)
                   .build(DatasetDefinitionRegistryFactory.class));
+
+        bind(UsageRegistry.class).toProvider(SystemUsageRegistryProvider.class);
```

Comment:
I think you'll still have to bind this to a specific key. Otherwise, all instances of UsageRegistry where this module is included will turn into SystemUsageRegistry.